### PR TITLE
[topic_state_monitor]add readme

### PR DESCRIPTION
**Pull request by tkimura4**
## Description
Add readme of topic_state_monitor

## Review Procedure
Please check the readme

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Read [pull request guidelines][pull-request-guidelines]
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: system/topic_state_monitor/Readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+The result of topic status is published as diagnostics.
+
+## Inner-workings / Algorithms
+
+The types of topic status and corresponding diagnostic status are following.
+
+| topic status  | diagnostic status | Description                                          |
+| ------------- | ----------------- | ---------------------------------------------------- |
+| `OK`          | OK                | The topic has no abnormalities                       |
+| `NotReceived` | ERROR             | The topic has not received yet.                      |
```
Comment (kmiya) [inline system/topic_state_monitor/Readme.md:15]:
```suggestion
| `NotReceived` | ERROR             | The topic has not been received yet                  |
```
Nit: add `been` and remove `.`.

File: system/topic_state_monitor/Readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+| Name     | Type     | Description                       |
+| -------- | -------- | --------------------------------- |
+| any name | any type | Subscribe target topic to monitor |
+
+### Output
+
+| Name           | Type                              | Description         |
+| -------------- | --------------------------------- | ------------------- |
+| `/diagnostics` | `diagnostic_msgs/DiagnosticArray` | diagnostics outputs |
```
Comment (kmiya) [inline system/topic_state_monitor/Readme.md:32]:
```suggestion
| `/diagnostics` | `diagnostic_msgs/DiagnosticArray` | Diagnostics outputs |
```
Nit: might be better to be upper case?

File: system/topic_state_monitor/Readme.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| `window_size` | int    | 10            | Window size of target topic for calculating frequency |
+
+### Core Parameters
+
+| Name              | Type   | Default Value | Description                                                                                          |
+| ----------------- | ------ | ------------- | ---------------------------------------------------------------------------------------------------- |
+| `topic`           | string | -             | Name of target topic                                                                                 |
+| `topic_type`      | string | -             | Type pf target topic                                                                                 |
+| `transient_local` | bool   | false         | QoS policy of topic Subscription (Transient Local/Volatile)                                          |
+| `best_effort`     | bool   | false         | QoS policy of topic Subscription (Best Effort/Reliable)                                              |
```
Comment (kmiya) [inline system/topic_state_monitor/Readme.md:50]:
```suggestion
| `transient_local` | bool   | false         | QoS policy of topic subscription (Transient Local/Volatile)                                          |
| `best_effort`     | bool   | false         | QoS policy of topic subscription (Best Effort/Reliable)                                              |
```
Nit: might be better to be lower case?

File: system/topic_state_monitor/Readme.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+| ------------- | ------ | ------------- | ----------------------------------------------------- |
+| `update_rate` | double | 10.0          | Timer callback period [Hz]                            |
+| `window_size` | int    | 10            | Window size of target topic for calculating frequency |
+
+### Core Parameters
+
+| Name              | Type   | Default Value | Description                                                                                          |
+| ----------------- | ------ | ------------- | ---------------------------------------------------------------------------------------------------- |
+| `topic`           | string | -             | Name of target topic                                                                                 |
+| `topic_type`      | string | -             | Type pf target topic                                                                                 |
```
Comment (kmiya) [inline system/topic_state_monitor/Readme.md:48]:
```suggestion
| `topic_type`      | string | -             | Type of target topic                                                                                 |
```
Nit: pf -> of

File: system/topic_state_monitor/Readme.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+## Purpose
+
+This node monitors input topic for abnormalities such as timeout and low frequency.
+The result of topic status is published as diagnostics.
+
+## Inner-workings / Algorithms
+
+The types of topic status and corresponding diagnostic status are following.
+
+| topic status  | diagnostic status | Description                                          |
```
Comment (kmiya) [inline system/topic_state_monitor/Readme.md:12]:
```suggestion
| Topic status  | Diagnostic status | Description                                          |
```
Nit: might be better to be upper case?

**Review by taikitanaka3 — APPROVED** [review]

**Pull request merged**